lib/spack/spack/build_systems/cmake.py

```diff
@@ -108,6 +108,11 @@ def _conditional_cmake_defaults(pkg: spack.package_base.PackageBase, args: List[
     if _supports_compilation_databases(pkg):
         args.append(CMakeBuilder.define("CMAKE_EXPORT_COMPILE_COMMANDS", True))
 
+    # Enable MACOSX_RPATH by default
+    # https://cmake.org/cmake/help/latest/policy/CMP0042.html
+    if pkg.spec.satisfies("platform=darwin") and cmake.satisfies("@3:"):
+        args.append(CMakeBuilder.define("CMAKE_POLICY_DEFAULT_CMP0042", "NEW"))
```

Comment:
Yeah, CMake has pretty good backward compat support. Any package that specifies `cmake_minimum_required` to something very low will get the default settings from the oldest version. (unless they set a policy max, but nobody does that).

This PR fixes the problem for all such packages instead of fixing them individually.

Reply:
Yup you got it. If the calling cmake is `@3:` but `cmake_minimum_required(2.8)` then CMake defaults to `CMAKE_POLICY_DEFAULT_CMP0042=OLD`.

Reply:
I have expanded on the comment to further clarify this